File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/game/context/mount/MountXpRatioMessage.scala





















// Generated on 05/03/2013 10:34:41
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait MountXpRatioMessage extends MessageDeserializer {
  val messageId = 5970

  protected trait Message extends BaseMessage with Serializable {
    val messageId = MountXpRatioMessage.messageId

    val ratio: Byte

    def serialize(buf: Buffer) {
      buf.writeByte(ratio)
    }

  }

  protected def create(ratio: Byte): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val ratio = buf.readByte()

    create(ratio)
  }

}

object MountXpRatioMessage extends MountXpRatioMessage {
  case class MessageImpl(ratio: Byte) extends Message
  type Target = MessageImpl

  override def create(ratio: Byte) = MessageImpl(ratio)

  def apply(ratio: Byte) = create(ratio)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

File: mambo-protocol/src/main/java/org/mambo/protocol/messages/MountXpRatioMessage.java


// Generated on 05/08/2013 19:37:45
package org.mambo.protocol.messages;

import java.util.*;
import org.mambo.protocol.types.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class MountXpRatioMessage extends NetworkMessage {
    public static final int MESSAGE_ID = 5970;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public byte ratio;
    
    public MountXpRatioMessage() { }
    
    public MountXpRatioMessage(byte ratio) {
        this.ratio = ratio;
    }
    
    @Override
    public void serialize(Buffer buf) {
        buf.writeByte(ratio);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        ratio = buf.readByte();
        if (ratio < 0)
            throw new RuntimeException("Forbidden value on ratio = " + ratio + ", it doesn't respect the following condition : ratio < 0");
    }
    
}
